src/full/Agda/Interaction/BasicOps.hs

```diff
@@ -1090,9 +1090,9 @@ introTactic pmLambda ii = do
      `catchError` \_ -> return []
     _ -> __IMPOSSIBLE__
   where
-    conName :: [NamedArg SplitPattern] -> [I.ConHead]
-    conName [p] = [ c | I.ConP c _ _ <- [namedArg p] ]
-    conName _   = __IMPOSSIBLE__
+    conHead :: [NamedArg SplitPattern] -> [I.ConHead]
```

Comment:
Maybe this function needn't be renamed (I think `conName` is a bit more telling for this function).
However, it's type could be made more precise:
```
... -> Maybe I.ConHead
```
It returns at most one constructor name (in form of a `ConHead` data structure).